src/main/java/org/nebula/contrib/ngbatis/session/IntervalCheckSessionDispatcher.java

```diff
@@ -0,0 +1,118 @@
+package org.nebula.contrib.ngbatis.session;
+
+// Copyright (c) 2022 All project authors and nebula-contrib. All rights reserved.
+//
+// This source code is licensed under Apache 2.0 License.
+
+import com.vesoft.nebula.client.graph.NebulaPoolConfig;
+import org.nebula.contrib.ngbatis.SessionDispatcher;
+import org.nebula.contrib.ngbatis.config.EnvConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.nebula.contrib.ngbatis.proxy.MapperProxy.ENV;
+
+/**
+ * 间隔时间进行检查的本地会话调度器
+ *
+ * @author yeweicheng
+ * @since 2022-08-26 2:34
+ * <br>Now is history!
+ */
+public class IntervalCheckSessionDispatcher implements Runnable, SessionDispatcher {
+
+    private static Logger log = LoggerFactory.getLogger(IntervalCheckSessionDispatcher.class);
+
+    private final NebulaPoolConfig nebulaPoolConfig;
+
+    private final ArrayBlockingQueue <LocalSession> sessionQueue;
+    private final ScheduledExecutorService threadPool;
+
+    public static long SESSION_LIFE_LENGTH = 5 * 60 * 60 * 1000;
+    public static long CHECK_FIXED_RATE = 5 * 60 * 60 * 1000; // unit ms
+
+    public IntervalCheckSessionDispatcher(NebulaPoolConfig nebulaPoolConfig) {
+        this.nebulaPoolConfig = nebulaPoolConfig;
+        this.sessionQueue = new ArrayBlockingQueue<>( nebulaPoolConfig.getMaxConnSize() );
+        threadPool = EnvConfig.reconnect ? Executors.newScheduledThreadPool(1) : null;
+        wakeUp();
+    }
+
+    @Override
+    public void run() {
+        for (LocalSession session : sessionQueue) {
+            log.info( "LocalSession in queue which created at {}, useCount: {}",
+                    session.getBirth(), session.useCount );
+
+            boolean finished = timeToRelease(session);
+            if( finished || !session.getSession().ping() ) {
+                log.info( "Release a session which created at {}", session.getBirth() );
+                release(session);
+            }
+        }
+        while (sessionQueue.size() < nebulaPoolConfig.getMinConnSize()) {
+            offer();
+        }
+    }
+
+    @Override
+    public void offer( LocalSession session ) {
+        boolean offer = sessionQueue.offer(session);
+        if(!offer) {
+            releaseInnerSession(session);
+        }
+    }
+
+    @Override
+    public LocalSession poll() {
+        LocalSession localSession = null;
+        if( !EnvConfig.reconnect ) {
+            localSession = newLocalSession();
+            localSession.useCount ++;

```

Comment:
if users use the executeWithParameter in mutil threads, how to keep the `poll` function thread-safe?

localSession.useCount++ is not safe here.

Reply:
Do you mean that different users use the same session to execute different nGQLs

Reply:
If I add synchronized to the poll method, can it help?

Reply:
Yeah I thank so.